cpp/velox/compute/VeloxPlanConverter.cc

```diff
@@ -35,183 +34,87 @@ VeloxPlanConverter::VeloxPlanConverter(
-  } else {
-    throw std::runtime_error("Child expected");
+namespace {
+std::shared_ptr<SplitInfo> parseScanSplitInfo(
+    const google::protobuf::RepeatedPtrField<substrait::ReadRel_LocalFiles_FileOrFiles>& fileList) {
+  using SubstraitFileFormatCase = ::substrait::ReadRel_LocalFiles_FileOrFiles::FileFormatCase;
+
+  auto splitInfo = std::make_shared<SplitInfo>();
+  splitInfo->paths.reserve(fileList.size());
+  splitInfo->starts.reserve(fileList.size());
+  splitInfo->lengths.reserve(fileList.size());
+  splitInfo->partitionColumns.reserve(fileList.size());
+  for (const auto& file : fileList) {
+    // Expect all Partitions share the same index.
+    splitInfo->partitionIndex = file.partition_index();
+
+    std::unordered_map<std::string, std::string> partitionColumnMap;
+    for (const auto& partitionColumn : file.partition_columns()) {
+      partitionColumnMap[partitionColumn.key()] = partitionColumn.value();
+    }
+    splitInfo->partitionColumns.emplace_back(partitionColumnMap);
+
+    splitInfo->paths.emplace_back(file.uri_file());
+    splitInfo->starts.emplace_back(file.start());
+    splitInfo->lengths.emplace_back(file.length());
+    switch (file.file_format_case()) {
+      case SubstraitFileFormatCase::kOrc:
+        splitInfo->format = dwio::common::FileFormat::ORC;
+        break;
+      case SubstraitFileFormatCase::kDwrf:
+        splitInfo->format = dwio::common::FileFormat::DWRF;
+        break;
+      case SubstraitFileFormatCase::kParquet:
+        splitInfo->format = dwio::common::FileFormat::PARQUET;
+        break;
+      case SubstraitFileFormatCase::kText:
+        splitInfo->format = dwio::common::FileFormat::TEXT;
+        break;
+      default:
+        splitInfo->format = dwio::common::FileFormat::UNKNOWN;
+        break;
+    }
   }
+  return splitInfo;
 }
 
-void VeloxPlanConverter::setInputPlanNode(const ::substrait::AggregateRel& sagg) {
-  if (sagg.has_input()) {
-    setInputPlanNode(sagg.input());
-  } else {
-    throw std::runtime_error("Child expected");
-  }
-}
+void parseLocalFileNodes(
+    SubstraitToVeloxPlanConverter* planConverter,
+    std::vector<::substrait::ReadRel_LocalFiles>& localFiles) {
+  std::vector<std::shared_ptr<SplitInfo>> splitInfos;
+  for (int32_t i = 0; i < localFiles.size(); i++) {
+    const auto& localFile = localFiles[i];
+    const auto& fileList = localFile.items();
 
-void VeloxPlanConverter::setInputPlanNode(const ::substrait::ProjectRel& sproject) {
-  if (sproject.has_input()) {
-    setInputPlanNode(sproject.input());
-  } else {
-    throw std::runtime_error("Child expected");
+    splitInfos.push_back(std::move(parseScanSplitInfo(fileList)));
   }
-}
 
-void VeloxPlanConverter::setInputPlanNode(const ::substrait::FilterRel& sfilter) {
-  if (sfilter.has_input()) {
-    setInputPlanNode(sfilter.input());
-  } else {
-    throw std::runtime_error("Child expected");
-  }
+  planConverter->setSplitInfos(std::move(splitInfos));
```

Comment:
Will we call `setSplitInfos` multiple times for an instance of planConverter?
https://github.com/oap-project/gluten/pull/4177#discussion_r1441506167

Reply:
should not, but not sure.